File: 0779-54-2658-maximum-number-of-fish-in-a-grid/0779-54-2658-maximum-number-of-fish-in-a-grid.cpp
class Solution {
public:
    

    int DFS(int x,int y,vector<vector<int>>& grid){
        if(x < 0 || x >= grid.size() || y  < 0 || y >= grid[0].size() || !grid[x][y])return 0;
        int ans = grid[x][y];
        grid[x][y] = 0;
        return DFS(x+1,y,grid) +  DFS(x-1,y,grid) +  DFS(x,y+1,grid)+  DFS(x,y-1,grid) + ans;
    }
    int findMaxFish(vector<vector<int>>& grid) {
        int mx = 0;
        for(int i = 0 ; i < grid.size();i++)
            for(int j = 0 ; j < grid[0].size();j++)
                if(grid[i][j])mx=max(mx,DFS(i,j,grid));
        return mx;
    }
};

File: 0779-54-2658-maximum-number-of-fish-in-a-grid/0779-54-2658-maximum-number-of-fish-in-a-grid.py
class Solution:
    def DFS(self, x, y, grid):
        if x < 0 or x >= len(grid) or y < 0 or y >= len(grid[0]) or grid[x][y] == 0:
            return 0
        ans = grid[x][y]
        grid[x][y] = 0
        return (
            ans
            + self.DFS(x + 1, y, grid)
            + self.DFS(x - 1, y, grid)
            + self.DFS(x, y + 1, grid)
            + self.DFS(x, y - 1, grid)
        )

    def findMaxFish(self, grid):
        mx = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] > 0:
                    mx = max(mx, self.DFS(i, j, grid))
        return mx
